src/bun.js/bindings/webcore/MessagePort.cpp

```diff
@@ -430,8 +430,7 @@ Ref<MessagePort> MessagePort::entangle(ScriptExecutionContext& context, Transfer
 bool MessagePort::addEventListener(const AtomString& eventType, Ref<EventListener>&& listener, const AddEventListenerOptions& options)
 {
     if (eventType == eventNames().messageEvent) {
-        if (listener->isAttribute())
-            start();
+        start();
```

Comment:
this change fixes the node:worker_threads module but it breaks cases using the raw `Worker` class.

Reply:
The current webkit code for MessagePort is designed to follow the Web API and can't handle by default event listener stacked registration on attribute

So, 
+ Revert the fix
+ Change the PR purpose: 
  - worker_threads TS module code cleanup
  - micro-optimization to webkit MessagePort event handling code  

I do not see why code cleanup should be lost 

A proper fix could be: 
+ Make a dedicated TS proxy class for worker_threads MessagePort and MessageChannel following/reusing the node EventEmitter API
+ Reuse it as parentPort in worker_threads module

The performance would be better if the implementation is part of the webkit glue. It's possible, the webkit EventEmitter class already support stacked listener registration 

In any case, that's the purpose of another PR.